pulsar-broker-common/src/main/java/org/apache/pulsar/broker/BaseGenerateDocumentation.java

```diff
@@ -80,50 +82,107 @@ public boolean run(String[] args) throws Exception {
+        ArrayList<Field> fieldList = new ArrayList<>(Arrays.asList(fields));
+
+        fieldList.removeIf(f -> f.getAnnotation(FieldContext.class) == null);
+        fieldList.sort(new CategoryComparator());
+        List<Field> requiredFields = fieldList.stream().filter(isRequired).toList();
+        List<Field> optionalFields = fieldList.stream().filter(isOptional).toList();
+        List<Field> deprecatedFields = fieldList.stream().filter(isDeprecated).toList();
 
         sb.append("# ").append(type).append("\n");
-        sb.append("|Name|Description|Default|Dynamic|Category|\n");
-        sb.append("|---|---|---|---|---|\n");
-        for (Field field : fields) {
-            FieldContext fieldContext = field.getAnnotation(FieldContext.class);
-            if (fieldContext == null) {
-                continue;
-            }
-            field.setAccessible(true);
-            sb.append("| ").append(field.getName()).append(" | ");
-            sb.append(fieldContext.doc().replace("\n", "<br>")).append(" | ");
-            sb.append(field.get(obj)).append(" | ");
-            sb.append(fieldContext.dynamic()).append(" | ");
-            sb.append(fieldContext.category()).append(" | ");
-            sb.append("\n");
-        }
+
+        sb.append(prefix).append(className).append(suffix);
+        sb.append("## Required\n");
+        writeDocListByFieldContext(requiredFields, sb, obj);
+        sb.append("## Optional\n");
+        writeDocListByFieldContext(optionalFields, sb, obj);
+        sb.append("## Deprecated\n");
+        writeDocListByFieldContext(deprecatedFields, sb, obj);
+
         return sb.toString();
     }
 
     protected String generateDocByApiModelProperty(String className, String type, StringBuilder sb) throws Exception {
         Class<?> clazz = Class.forName(className);
         Object obj = clazz.getDeclaredConstructor().newInstance();
         Field[] fields = clazz.getDeclaredFields();
+        ArrayList<Field> fieldList = new ArrayList<>(Arrays.asList(fields));
+
+        fieldList.sort(Comparator.comparing(Field::getName));
 
         sb.append("# ").append(type).append("\n");
-        sb.append("|Name|Description|Default|\n");
-        sb.append("|---|---|---|\n");
-        for (Field field : fields) {
+        sb.append(prefix).append(className).append(suffix);
+        sb.append("## Optional\n");
+        for (Field field : fieldList) {
             ApiModelProperty fieldContext = field.getAnnotation(ApiModelProperty.class);
             if (fieldContext == null) {
                 continue;
             }
             field.setAccessible(true);
+
             String name = StringUtils.isBlank(fieldContext.name()) ? field.getName() : fieldContext.name();
-            sb.append("| ").append(name).append(" | ");
-            sb.append(fieldContext.value().replace("\n", "<br>")).append(" | ");
-            sb.append(field.get(obj)).append(" | ");
-            sb.append("\n");
```

Comment:
Won't these changes affect the generation of other documents?

Reply:
@urfreespace No, I've checked all of the above and they won't affect other docs.